pkg/wasm/cache.go

```diff
@@ -52,7 +53,7 @@ const (
 
 // Cache models a Wasm module cache.
 type Cache interface {
-	Get(url, checksum string, timeout time.Duration, pullSecret []byte) (string, error)
+	Get(url, checksum, resourceName, resourceVersion string, timeout time.Duration, pullSecret []byte, pullPolicy extensions.PullPolicy) (string, error)
```

Comment:
4 strings/7 parameters is a lot.. maybe we should extract to a struct?

Reply:
You are right. 

I am thinking about some sort of refactoring around the cache and fetchers after implementing the feature and integration tests.